### ISSUE TITLE
Fatal crash / Error

### ISSUE THREAD

**Issue by EvilTactician** [IUrreta/DatabaseEditor#90]
Change made:

(McLarenn)
Swap Piastri to Reserve Driver
Fire Palou
Hire Verschoor (F2)

First time I tried it, after FP2, a crash.
This was close to a race weekend, so I tried again right after a race weekend, to ensure this wasn't the problem. Next race went fine, then came a sprint weekend, essentially making qualifying the second session. This time, it crashed in qualifying.

Fatal error!

Unhandled Exception: EXCEPTION_ACCESS_VIOLATION reading address 0x0000000000000028

0x0000000141dbceea F1Manager23.exe!UnknownFunction []
0x0000000141db82be F1Manager23.exe!UnknownFunction []
0x0000000141dd0a90 F1Manager23.exe!UnknownFunction []
0x0000000141e37c83 F1Manager23.exe!UnknownFunction []
0x00000001543ae1d3 F1Manager23.exe!UnknownFunction []
0x0000000153dcd77a F1Manager23.exe!UnknownFunction []
0x000000014228feb9 F1Manager23.exe!UnknownFunction []
0x00000001423654e2 F1Manager23.exe!UnknownFunction []
0x0000000140b44050 F1Manager23.exe!UnknownFunction []
0x0000000141dc84ae F1Manager23.exe!UnknownFunction []
0x0000000142032292 F1Manager23.exe!UnknownFunction []
0x000000014204274f F1Manager23.exe!UnknownFunction []
0x0000000143d9a35b F1Manager23.exe!UnknownFunction []
0x00000001444dcbc7 F1Manager23.exe!UnknownFunction []
0x00000001420ac79e F1Manager23.exe!UnknownFunction []
0x00000001420af956 F1Manager23.exe!UnknownFunction []
0x00000001444f10e2 F1Manager23.exe!UnknownFunction []
0x00000001444f5db7 F1Manager23.exe!UnknownFunction []
0x00000001440b10d8 F1Manager23.exe!UnknownFunction []
0x0000000143f26d1c F1Manager23.exe!UnknownFunction []
0x0000000140b15be4 F1Manager23.exe!UnknownFunction []
0x0000000140b1f89c F1Manager23.exe!UnknownFunction []
0x0000000140b1f91a F1Manager23.exe!UnknownFunction []
0x0000000140b2068c F1Manager23.exe!UnknownFunction []
0x0000000140b29444 F1Manager23.exe!UnknownFunction []
0x000000014505dad2 F1Manager23.exe!UnknownFunction []
0x00007ffbfde626ad KERNEL32.DLL!UnknownFunction []
0x00007ffbfee6aa68 ntdll.dll!UnknownFunction []

In both cases, reverting to the save pre-edit, would make things work fine and get past those sessions. It looks like something gets corrupted when changing drivers?

[autosave.zip](https://github.com/IUrreta/DatabaseEditor/files/12446668/autosave.zip)

I've attached the save for your convenience.

> [2023-08-26 16:38:41.419994] INFO: Command executed: connect
[2023-08-26 16:38:46.003958] INFO: Command executed: saveSelected save1.sav
[2023-08-26 16:38:55.795213] INFO: Command executed: swap 102 285
[2023-08-26 16:38:58.715330] INFO: Command executed: 
[2023-08-26 16:39:10.118672] INFO: Command executed: fire 285
[2023-08-26 16:39:57.256810] INFO: Command executed: fire 99
[2023-08-26 16:39:57.384460] INFO: Command executed: hire 99 2 1 1000000 50000 100000 10 2028

**Comment by EvilTactician** [comment]
Not sure if related, but 1.5.4 inflates the save file size significantly, whilst 1.4.3 does not.

**Comment by IUrreta** [comment]
1. As long as I am concerned, that crash has nothing to do with the editor. The commands look good and that should not be the cause. Game crashes sometimes. 

2. For the second comment, 1.5.4 uses the extact same mechanism as 1.4.3 for repacking the save, so I don't know why it happened but if it happened in 1.5.4 it could happen as well as 1.4.3

**Comment by EvilTactician** [comment]
Let me elaborate then.

This crash only occurs after making the change highlighted in my report. It occurs at a specific interval/session, and using the save game without the edit progresses fine without a crash.
The *only* crashes I've had were all directly related to this. Use the editor to swap in the F2 driver, the game crashes within 2 race weekends.

The crash is consistently **at the same time**. (end of 2nd session)

I've tried the same edit on a clean save which I know could progress beyond the session, and it crashed again with the same error code. Without the edit, the game once again progresses fine. This is most definitely related specifically to one of those driver swaps, and I suspect it is the F2 driver.

It doesn't look like the game is properly handling him having vanished from his team. It appears to work fine for F1 driver swaps and changes.

You're more than welcome to ignore this -- but this report is to help **you**.

**Comment by IUrreta** [comment]
If that's the cas then I'll look into it, but I have tried both in 22 and 23 multiple times transfers with F2, F3, retired and regen drivers and none of them crashed, that's why I thought it was not related with the editor. In fact, I would have not released it knowing that transfering F2 drivers causes a crash, as this is the first report I get of constant crashes because of one transfer. Thanks for the info.

**Comment by EvilTactician** [comment]
> If that's the cas then I'll look into it, but I have tried both in 22 and 23 multiple times transfers with F2, F3, retired and regen drivers and none of them crashed, that's why I thought it was not related with the editor. In fact, I would have not released it knowing that transferring F2 drivers causes a crash, as this is the first report I get of constant crashes because of one transfer. Thanks for the info.

The most complicated part is that the crash doesn't occur until the end of the 2nd session up to 2 races after. (At least so far?)

I can only speculate on what may be the cause, perhaps that's when F2 had an event, which then caused the crash due to issues with the driver. I hope the save is useful, it happens in that save after the 2nd session.

I've got around it (lost about 6 hours play time, but that's life, I just hope it helps find a potential bug!) this way:

- Reverted to backed up save. (I manually backed up, didn't realise you added a feature recently!)
- Manually in-game offered RVE a contract.
- Edited his contract to something not completely stupid via your editor. Contract negotiations being stupid was the main reason I even found your editor... I don't like that I can give someone a big sign-on bonus to massively reduce their annual salary, it's completely unrealistic. Why would someone give up 5M over a few year to get 100K now? Unless they have gambling debts...
- Everything works fine.

What I am not 100% sure about is:

- I swapped Palou (reserve) with Piastri so Piastri became reserve
- I then swapped Verschoor into Palou (who was now driver 1)
- If I had just done Verschoor directly to Piastri, would it have been fine?

Either way it would be a bug, but some weird interaction going on possibly?

For the record, loving your work. This is a very slick editor with a very user friendly UI.
